### PR TITLE
NEXUS-8062: Delete npm metadata on repo removal

### PR DESCRIPTION
**Pull request by cstamas**
Introduce delete on repository removal.

Issue
https://issues.sonatype.org/browse/NEXUS-8062

CI
http://bamboo.s/browse/NX-OSSF564

### CODE REVIEW

**Comment by jdillon** [comment]
+0 really no idea, but if this makes it work I'll +1

**Comment by mcculls** [comment]
+1 once 2.11.2 is out-the-door

File: plugins/npm/nexus-npm-repository-plugin/src/main/java/com/bolyuba/nexus/plugin/npm/hosted/DefaultNpmHostedRepository.java [hunk 1]
```diff
@@ -152,38 +163,47 @@ protected AbstractStorageItem doRetrieveLocalItem(ResourceStoreRequest storeRequ
...
       }
-      PackageRequest packageRequest = new PackageRequest(storeRequest);
-      if (packageRequest.isMetadata()) {
-        ContentLocator contentLocator;
-        if (packageRequest.isRegistryRoot()) {
-          contentLocator = hostedMetadataService.produceRegistryRoot(packageRequest);
-        }
-        else if (packageRequest.isPackageRoot()) {
-          contentLocator = hostedMetadataService.producePackageRoot(packageRequest);
+      PackageRequest packageRequest = null;
```
Comment (kellyrob99) [inline plugins/npm/nexus-npm-repository-plugin/src/main/java/com/bolyuba/nexus/plugin/npm/hosted/DefaultNpmHostedRepository.java]: any way to share this duplicated code?

**Comment by kellyrob99** [comment]
+1

File: plugins/npm/nexus-npm-repository-plugin/src/main/java/com/bolyuba/nexus/plugin/npm/hosted/DefaultNpmHostedRepository.java (hunk 1 above)
Comment (cstamas) [inline plugins/npm/nexus-npm-repository-plugin/src/main/java/com/bolyuba/nexus/plugin/npm/hosted/DefaultNpmHostedRepository.java]: Both, retrieve and delete methods of hosted npm and proxy npm repositories share a lot of similarities, if not almost equal. Still, as original implementation went with npm hosted repository extending core's AbstractRepository and npm proxy repository extending core's AbstractProxyRepository, this duplication was implicit result of it. NX2 Maven repositories were implemented differently (they both, M2 hosted and M2 proxy is extended from AbstractProxyRepository, and can share common ancestor, AbstractMavenRepository). This duplication is result of problematic core classes and deep hierachy, and will be gone with NX3 repo API hopefully. I think it's OK to leave NX2 to live with it for now.

Still, comments added to point out this fact to next developer wandering in this area.

**Comment by cstamas** [comment]
Merged as 6c4976202c03d78e595d6168ef870fb35943f4bc
(with changes to point out code duplication)

**Pull request closed without merging**